File: go/internal/fizzbuzz.go
package internal

import (
	"strconv"
)

func GenerateFizzBuzz(int1, int2, limit uint, str1, str2 string) []string {
	result := make([]string, limit)

	for i := uint(1); i <= limit; i++ {
		div1 := i%int1 == 0
		div2 := i%int2 == 0

		switch {
		case div1 && div2:
			result[i-1] = str1 + str2
		case div1:
			result[i-1] = str1
		case div2:
			result[i-1] = str2
		default:
			result[i-1] = strconv.FormatUint(uint64(i), 10)
		}
	}
	return result
}

File: python/internal/fizzbuzz.py
def generate_fizz_buzz(
    int1: int, int2: int, limit: int, str1: str, str2: str
) -> list[str]:
    result: list[str] = []

    for i in range(1, limit + 1):
        s = ""
        if not i % int1:
            s += str1
        if not i % int2:
            s += str2
        result.append(s or str(i))

    return result
